header.go

```diff
@@ -2133,6 +2178,35 @@ func getHeaderKeyBytes(kv *argsKV, key string, disableNormalizing bool) []byte {
 	return kv.key
 }
 
+func normalizeHeaderValue(buffer []byte) (value []byte) {
+	length := len(buffer)
+	if length <= 0 {
+		value = []byte{}
+		return
+	}
+	value = make([]byte, length)
```

Comment:
This function now allocates in the hot path which we really can't have.

I think the best way to fix this is to rewrite the input (`headerScanner.b`) to remove the `\r\n` within the header value. This means future parses of the header buffer will still work. But this does require some modifications to your code. Instead of calling `normalizeHeaderValue` you will need to move everything within `headerScanner.b` back by 2 characters instead.

You can add this test to `allocation_test.go` to make sure it works:
```go
func TestAllocationHeaderScanner(t *testing.T) {
  t.Parallel()

  rawHeaders := []byte("HTTP/1.1 200 OK\r\n" +
    "Date: Mon, 04 Nov 2019 03:08:05 GMT\r\n" +
    "Transfer-Encoding: chunked\r\n" +
    "Content-Type: multipart/related; type=\"application/xop+xml\";\r\n" +
    " boundary=\"----=_Part_16_1220614102.1572836885552\"\r\n" +
    "\r\n" +
    "body\r\n")

  n := testing.AllocsPerRun(100, func() {
    var s headerScanner
    s.b = rawHeaders
    for s.next() {
    }
  })

  if n != 0 {
    t.Fatalf("expected 0 allocations, got %f", n)
  }
}
```

Another test you can add to make sure things keep working is a test that calls `RequestHeader.VisitAllInOrder()` twice and see if the headers on the second call are still correct.